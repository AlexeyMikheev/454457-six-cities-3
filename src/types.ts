import {OfferType} from "./consts";

export interface Location {
    latitude: number;
    longitude: number;
    zoom: number;
    center: number[];
}

export interface City {
    name: string;
    location: Location;
    center: number[];
    zoom: number;
}

export interface Owner {
    id: number;
    name: string;
    avatar: string;
    isTrust: boolean;
}

export interface Offer {
    owner: Owner;
    location: Location;
    city: City;
    id: number;
    isPremium: boolean;
    isMarked: boolean;
    cost: number;
    rating: number;
    name: string;
    description: string;
    roomsCount: number;
    membersCount: number;
    type: OfferType;
    image: string;
    images: string[];
    features: string[];
    cityName: string;
    center: number[];
    zoom: number;
}

export interface UserInfo {
  avatar_url: string;
  email: string;
  id: number;
  is_pro: boolean;
  name: string;
}

export interface Review{
    id: number;
    comment: string;
    rating: number;
    date: number;
    user: Owner;
  }

export type HoveredChangeArgType = number | null;
export type SendCommentArgType = {
    rating: string;
    comment: string;
}
export type FormValueArgsType = {
    target: {
        name: string;
        value: string;
    };
}

export type SubmitArgsType = {
    login: string;
    password: string;
}

export interface FormState{
    [key: string]: string;
 }

export interface BooleanState{
    isToggled: boolean;
 }

export type OfferHoveredChangeFunction = (offerId: HoveredChangeArgType) => void | null;
export type OfferChangeFunction = (offerId: number) => void;
export type CityChangeFunction = (cityName: string) => void;
export type SendCommentFunction = (offerId: number, args: SendCommentArgType) => void;
export type FormValueChangeFunction = (args: FormValueArgsType) => void;
export type FormSubmitFunction = (args: SubmitArgsType) => void;

export type EmptyFunction = () => void;

export type MapSetting = {
    zoomControl: boolean;
    marker: boolean;
}
