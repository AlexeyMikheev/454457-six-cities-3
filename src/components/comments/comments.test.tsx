import * as React from "react";
import {mount} from "enzyme";
import Comments from "./comments";
import {Review} from '../../types';

const mockDate: number = new Date(0).valueOf();

const mocksReviews: Review[] = [
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: mockDate,
    id: 1,
    rating: 4,
    user: {
      avatar: `img/1.png`,
      id: 4,
      isTrust: false,
      name: `Max`
    }
  },
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: mockDate,
    id: 2,
    rating: 4,
    user: {
      avatar: `img/1.png`,
      id: 4,
      isTrust: false,
      name: `Max`
    }
  },
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: mockDate,
    id: 3,
    rating: 4,
    user: {
      avatar: `img/1.png`,
      id: 4,
      isTrust: false,
      name: `Max`
    }
  }
];

it(`Comments snapshot`, () => {
  const div = document.createElement(`div`);
  document.body.appendChild(div);

  const commentsComponent = mount(<Comments reviews={mocksReviews} />, {attachTo: div});

  expect(commentsComponent.getDOMNode()).toMatchSnapshot();
});
