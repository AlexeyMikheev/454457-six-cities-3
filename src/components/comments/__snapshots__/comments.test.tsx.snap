// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments snapshot 1`] = `
Array [
  <h2
    class="reviews__title"
  >
    Reviews Â· 
    <span
      class="reviews__amount"
    >
      3
    </span>
  </h2>,
  <ul
    class="reviews__list"
  >
    <li
      class="reviews__item"
    >
      <div
        class="reviews__user user"
      >
        <div
          class="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            class="reviews__avatar user__avatar"
            height="54"
            src="img/1.png"
            width="54"
          />
        </div>
        <span
          class="reviews__user-name"
        >
          Max
        </span>
      </div>
      <div
        class="reviews__info"
      >
        <div
          class="reviews__rating rating"
        >
          <div
            class="reviews__stars rating__stars"
          >
            <span
              style="width: 80%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          class="reviews__text"
        >
          A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
        </p>
        <time
          class="reviews__time"
          datetime="2019-04-24"
        >
          January Th, 1970
        </time>
      </div>
    </li>
    <li
      class="reviews__item"
    >
      <div
        class="reviews__user user"
      >
        <div
          class="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            class="reviews__avatar user__avatar"
            height="54"
            src="img/1.png"
            width="54"
          />
        </div>
        <span
          class="reviews__user-name"
        >
          Max
        </span>
      </div>
      <div
        class="reviews__info"
      >
        <div
          class="reviews__rating rating"
        >
          <div
            class="reviews__stars rating__stars"
          >
            <span
              style="width: 80%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          class="reviews__text"
        >
          A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
        </p>
        <time
          class="reviews__time"
          datetime="2019-04-24"
        >
          January Th, 1970
        </time>
      </div>
    </li>
    <li
      class="reviews__item"
    >
      <div
        class="reviews__user user"
      >
        <div
          class="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            class="reviews__avatar user__avatar"
            height="54"
            src="img/1.png"
            width="54"
          />
        </div>
        <span
          class="reviews__user-name"
        >
          Max
        </span>
      </div>
      <div
        class="reviews__info"
      >
        <div
          class="reviews__rating rating"
        >
          <div
            class="reviews__stars rating__stars"
          >
            <span
              style="width: 80%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          class="reviews__text"
        >
          A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
        </p>
        <time
          class="reviews__time"
          datetime="2019-04-24"
        >
          January Th, 1970
        </time>
      </div>
    </li>
  </ul>,
]
`;
